### PR TITLE
Move all logging to log4j.

### PR DESCRIPTION
**Pull request by cmnbroad**
This moves all logging to log4j (previously we had a mix of log4j and SAMTools logging). 

Questions:There are about 35 places where we were using the SAMTools ProgressLogger, which assumes SAMTools logging. I reproduced that class in hellbender, but implemented the SAMTools ProgressLoggerInterface in order to retain compatibility with SAMWriters, since we use those in a couple of places. The hellbender ProgressLogger class is in utils.runtime, not sure if there is a better place for it. Also I'm not sure how to handle the source code attribution of  it since it was lifted from SAMTools but slightly modified for hellbender ? Can I do that or do I need to rewrite it from scratch ?

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/picard/analysis/CollectAlignmentSummaryMetrics.java [hunk 1]
```diff
@@ -45,7 +46,7 @@
 public final class CollectAlignmentSummaryMetrics extends SinglePassSamProgram {
     static final String USAGE = "Produces from a SAM or BAM a file containing summary alignment metrics";
     
-    private static final Log log = Log.getInstance(CollectAlignmentSummaryMetrics.class);
+    private static final Logger log = LogManager.getLogger();
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/picard/analysis/CollectAlignmentSummaryMetrics.java]: should this take the class name?

File: src/main/java/org/broadinstitute/hellbender/tools/picard/analysis/CollectOxoGMetrics.java [hunk 2]
```diff
@@ -229,7 +228,7 @@ protected Object doWork() {
         }
         iterator.setSamFilters(filters);
 
-        log.info("Starting iteration.");
+        logger.info("Starting iteration.");
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/picard/analysis/CollectOxoGMetrics.java]: What's the rational for using the base class logger here but declaring a new one in other places?  I think we should probably prefer declaring a new one in every specific class so the messages are more specific, but I suspect the existing code is just not consistent about this.

**Comment by lbergelson** [comment]
I refactored ProgressLogger and I'm pushing it up for review.  (See https://github.com/samtools/htsjdk/pull/281.)  I will feel much happier just implementing a `log` function than copying that code over and having to test and maintain it.

**Comment by lbergelson** [comment]
@cmnbroad Going to get back to this once I hear from htsjdk folk about the proposed change.  Let me know if you think it doesn't make sense.

**Comment by cmnbroad** [comment]
Great - thanks for doing that.

**Comment by lbergelson** [comment]
@cmnbroad We're waiting on the next htsjdk release now.  Should be monday.

**Comment by lbergelson** [comment]
This is currently held up by #758 which I'm working on now.

**Comment by lbergelson** [comment]
@cmnbroad We've merged the htsjdk changes!   Could you refactor the progress logger by extending the new `AbstractProgressLogger`.  Thanks and sorry for the delay.

**Comment by cmnbroad** [comment]
I moved over to the AbstractLoggerInterface (@lb - thanks for getting that through). Ideally we'd merge this PR before we take #603 since once we take this one I should be able to remove the code from 603 that propagates the VERBOSITY level to Picard.

File: src/main/java/org/broadinstitute/hellbender/tools/picard/sam/BamToBfq.java [hunk 3]
```diff
@@ -0,0 +1,492 @@
+package org.broadinstitute.hellbender.tools.picard.sam;
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/picard/sam/BamToBfq.java]: I think this should be deleted.  Must have been deleted in master and re-added by accident.

File: src/main/java/org/broadinstitute/hellbender/utils/runtime/ProgressLogger.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    public ProgressLogger(final Logger logger) { this(logger, 1000000); }
+
+    /**
+     * Log a message to whatever logger is being used
+     *
+     * @param message a message to be logged by the logger (recommended output level is INFO or the equivalent)
+     */
+    protected void log(String... message) {
+        logger.info(message);
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/utils/runtime/ProgressLogger.java]: Does this work with the varargs input?

**Comment by lbergelson** [comment]
@cmnbroad Looks good, just kill that file that snuck back in and assure me that  the progress logger actually outputs messages and not some variant of `StringArray[10929123@asdfa]` :)

**Comment by cmnbroad** [comment]
Changed the varargs code, and made a small change to the ProgressLogger implementation to fix another problem I found while testing. Also re-deleted the file you mentioned. Thx for finding those.

**Comment by lbergelson** [comment]
:+1:

**Comment by lbergelson** [comment]
@cmnbroad Thanks :)

**Pull request merged**